js/test/test.js

```diff
@@ -478,6 +564,35 @@ let testBalance = async (exchange, symbol) => {
 
 //-----------------------------------------------------------------------------
 
+let testNonce = async (exchange, symbol) => {
+    log.green ('AuthenticationError test...')
+    let nonce = exchange.nonce
+    exchange.nonce = () => 1
+    try {
+        if (exchange.hasFetchBalance || exchange.has.fetchBalance)
+            await exchange.fetchBalance ();
+        else if (exchange.hasFetchMyTrades || exchange.has.fetchMyTrades)
+            await exchange.fetchMyTrades (symbol, 0)
+        else if (exchange.hasFetchOrders || exchange.has.fetchOrders)
+            await exchange.fetchOrders (symbol)
+        else
+            exchange.nonce = nonce
+            return
```

Comment:
Line 580 shows that you got used to Python scripting too much ))
The next line 581 is unreachable from here )

Reply:
No worries, I'm already merging it )

Reply:
Ouch! Indeed )